### PR TITLE
Feature/109 skype bot support

### PR DESCRIPTION
**Pull request by Awarua-**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,6 +101,7 @@ Now you can customize your Concierge by installing or creating your own modules.
 		- [Skype Integration](doc/integrations/Skype.md)
 		- [Telegram Integration](doc/integrations/Telegram.md)
 		- [Messenger Bots](doc/integrations/Messenger.md)
+        - [MSBot](doc/integrations/MSBot.md)
```
Comment (mrkno) [inline README.md]: Inconsistent indentation

File: core/core_integrations/MSBot.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
+let fs = require('fs'),
+    restify = require('restify'),
+    builder = require('botbuilder'),
+    entities = require('entities'),
+    mime = require('mime'),
+    callback = null,
+    server = null,
+    botService = null,
+    api = null,
+    locale = $$.getLocale(),
```
Comment (mrkno) [inline core/core_integrations/MSBot.js]: locale can change at runtime, should just be called as needed

File: core/core_integrations/MSBot.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        return userCache;
+    },
+
+    receiveMessage = (session) => {
+        if (!bot[session.message.source]) {
+            bot[session.message.source] = session.message.address.bot;
+        }
+
+        if (!userCache[session.message.address.user.id]) {
+            userCache[session.message.address.user.id] = session.message.address.user.name
```
Comment (mrkno) [inline core/core_integrations/MSBot.js]: ;

File: core/core_integrations/MSBot.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    const port = config.port || 8000;
+    server.listen(port, () => {
+        console.debug(`${server.name} listening to ${server.url}`);
+    });
+
+    let connector = new builder.ChatConnector({ appId: config.app_id, appPassword: config.app_secret_password });
+    botService = new builder.UniversalBot(connector);
+
+    api = shim.createIntegration({
+		sendMessage: sendMessage,
```
Comment (mrkno) [inline core/core_integrations/MSBot.js]: inconsistent indentation

File: core/core_integrations/MSBot.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        getUsers: getUsers
+    });
+
+    server.post('/api/messages', connector.listen());
+
+    let intents = new builder.IntentDialog();
+    botService.dialog('/', intents);
+
+    intents.onDefault([
+        function (session, results) {
```
Comment (mrkno) [inline core/core_integrations/MSBot.js]: es5 in mostly es6 code?

**Comment by mrkno** [comment]
Btw, rename MSBot.js to msbot.js. We are case sensitive

File: doc/integrations/MSBot.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+			"commandPrefix": "!"
+		}
+	}
+}
+```
+Replace each of the angle bracketed strings (`<...>`) with the respective information written inside the brackets.
+
+Notes:
+- To obtain a TLS certificate easily, the Lets Encrypt project has [easy ways of doing this](https://letsencrypt.org/getting-started/).
+- Currently MSBots works on platforms supported [here]().
```
Comment (mrkno) [inline doc/integrations/MSBot.md]: Incomplete documentation

File: doc/integrations/MSBot.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+	"output": {
+		"msbot": {
+			"key": "<path to TLS key, eg key.pem>",
+			"cert": "<path to TLS cert, eg cert.pem>",
+			"app_id": "<Microsoft App ID>",
+			"app_secret_password": "<Microsoft App Secret Password>",
+			"port": <port>,
+			"commandPrefix": "!"
+		}
+	}
```
Comment (mrkno) [inline doc/integrations/MSBot.md]: Inconsistent indentation

File: package.json [hunk 8]
```diff
@@ -20,6 +21,7 @@
     "node-telegram-bot-api": "^0.23.3",
     "redwrap": "*",
     "request": "^2.72.0",
+    "restify": "^4.1.1",
```
Comment (mrkno) [inline package.json]: Are these not installed automatically as needed?

File: README.md [hunk 9]
```diff
@@ -101,7 +101,8 @@ Now you can customize your Concierge by installing or creating your own modules.
 		- [Skype Integration](doc/integrations/Skype.md)
 		- [Telegram Integration](doc/integrations/Telegram.md)
 		- [Messenger Bots](doc/integrations/Messenger.md)
-		- [Testing Mode](doc/integrations/Testing.md)
+ 		- [MSBot](doc/integrations/MSBot.md)
```
Comment (mrkno) [inline README.md]: inconsistent indentation

Comment (Awarua-) [inline README.md]: Fixed?

File: doc/integrations/MSBot.md (hunk 7 above)
Comment (Awarua-) [inline doc/integrations/MSBot.md]: ??

Comment (Awarua-) [inline doc/integrations/MSBot.md]: Looks fine to me

File: package.json (hunk 8 above)
Comment (Awarua-) [inline package.json]: Removed.
Tested and they are install automatically

File: doc/integrations/MSBot.md (hunk 6 above)
Comment (Awarua-) [inline doc/integrations/MSBot.md]: Fixed

**Comment by Awarua-** [comment]
Fixes #109

**Pull request merged**